connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -164,7 +165,7 @@ private[sql] class SparkConnectClient(
   }
 
   def copy(): SparkConnectClient = {
-    new SparkConnectClient(userContext, channel, userAgent)
+    SparkConnectClient.builder().connectionString(connectString).build()
```

Comment:
Only works if you only use a connection string right? So what about SSL, or when the host and port are set manually? Can't we just pass the ChannelBuilder around?

Reply:
If you see the current SparkSession builder, I think it only has two versions to build the SparkSession:

1. with ConnectionString.
2. nothing but default build.

that is why I only keep ConnectString. But If you think keep the builder to preserve everything is better I can update.

Reply:
https://github.com/apache/spark/blob/a77bb37b4112543fcd77a7f6091e465daeb3f8ae/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala#L497

Reply:
I don't buy that argument. The problem is that the client builder supports all these things (for good reason), now we are making the assumption that this thing will only be used to create a SparkSession, while there are also other use cases.

Please keep the builder.

Reply:
updated to use the builder thus we can copy entire channel setup to new client.